bionic/task_state.py

```diff
@@ -32,27 +34,44 @@ def __init__(self, task, dep_states, case_key, provider):
         # This can be set by get_results_assuming_complete() or _compute().
         self._results_by_name = None
 
+        # A completed task state has it's results computed and cached somewhere for
+        # easy retrieval.
         self.is_complete = False
 
+    # FIXME: We need a coherent story around incomplete states between parallel
+    # and non-parallel mode. We have to compute non-persisted deps inside the
+    # other subproces for parallel mode but not for non-parallel mode. We should
+    # consider making the way it computes in parallel mode the default. This
+    # would also match the definition of is_computable.
     def incomplete_dep_states(self):
         return [dep_state for dep_state in self.dep_states if not dep_state.is_complete]
 
     @property
-    def is_blocked(self):
-        return len(self.incomplete_dep_states()) > 0
+    def is_computable(self):
+        """
+        Indicates whether the task state can be computed without any additional
```

Comment:
"any additional setup" is vague and not really true, right?

Also, maybe we should include a note explaining why this definition is so complicated.

Reply:
Done